platform/trace.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "base/macros.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace platform
+{
+class TraceImpl;
+
+// API is inspired by https://developer.android.com/ndk/reference/group/tracing
+class Trace 
+{
+public:
+  static Trace & Instance() noexcept;
+
+  void BeginSection(char const * name) noexcept;
+  void EndSection() noexcept;
+  void SetCounter(char const * name, int64_t value) noexcept;
+
+private:
+  Trace();
+  ~Trace();
+  
+  std::unique_ptr<TraceImpl> m_impl;
+  
+  DISALLOW_COPY_AND_MOVE(Trace);
+};
+
+class TraceSection
+{
+public:
+  inline TraceSection(char const * section) noexcept
+  {
+    Trace::Instance().BeginSection(section);
+  }
+
+  inline ~TraceSection() noexcept
+  {
+    Trace::Instance().EndSection();
+  }
+};
+}  // namespace platform
+
+#ifdef ENABLE_TRACE
```

Comment:
Adding ```&& OMIM_OS_ANDROID``` here allows remove trace_empty.cpp and discard the project.pbxproj changes.